### PR TITLE
Extend usage to asymmetric ranges.

### PR DESCRIPTION
**Pull request by dhevang**
BEGINRELEASENOTES
- Extend usage of dumpBfield to asymmetric X,Y,Z ranges.

ENDRELEASENOTES

### CODE REVIEW

File: UtilityApps/src/dumpBfield.cpp [hunk 1]
```diff
@@ -41,12 +41,43 @@ static int invoke_dump_B_field(int argc, char** argv ){
...
-  zRange *= dd4hep::cm;
+  sstr >> RangeX >> RangeY >> RangeZ >> dx >> dy >> dz;
+
+  size_t colon_posX = RangeX.find(':');
+  size_t colon_posY = RangeY.find(':');
+  size_t colon_posZ = RangeZ.find(':');
+
+  float minX=0, maxX=0, minY=0, maxY=0, minZ=0, maxZ=0;
+
+  if( colon_posX == std::string::npos || colon_posY == std::string::npos || colon_posZ == std::string::npos ) {
```
Comment (andresailer) [inline UtilityApps/src/dumpBfield.cpp:55]: What about splitting this for each axis, then one could mix symmetric and asymetric limit setting?

File: UtilityApps/src/dumpBfield.cpp [hunk 2]
```diff
@@ -29,8 +29,8 @@ using namespace dd4hep::detail;
 static int invoke_dump_B_field(int argc, char** argv ){
   
   if( argc != 8 ) {
-    std::cout << " usage: dumpBfield compact.xml x y z dx dy dz [in cm]" << std::endl 
-	      << "    will dump the B-field in volume [-x:x,-y:y,-z,z] with steps [dx,dy,dz] "
+    std::cout << " usage: dumpBfield compact.xml xmin:xmax ymin:ymax zmin:zmax dx dy dz [in cm]" << std::endl 
+	      << "    will dump the B-field in volume [xmin:xmax, ymin:ymax, zmin:zmax] with steps [dx,dy,dz] "
```
[the remark this answers is not in the document]
Comment (dhevang) [reply UtilityApps/src/dumpBfield.cpp:33]: Sounds good.

File: UtilityApps/src/dumpBfield.cpp (hunk 1 above)
Comment (dhevang) [reply UtilityApps/src/dumpBfield.cpp]: Sure.  I committed the change.

**Review by andresailer — APPROVED** [review]
Thanks!

**Pull request merged**